File: WPF/MVVM/Converters/MultiplyMultiConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace Common.MVVM.Converters

    Public Class MultiplyMultiConverter
        Implements IMultiValueConverter

        Public Function Convert(values() As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
            Dim R = If(parameter Is Nothing, 1, CType(parameter, Double))
            For I = 0 To values.Length - 1
                R *= CType(values(I), Double)
            Next
            Return R
        End Function

        Public Function ConvertBack(value As Object, targetTypes() As Type, parameter As Object, culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Throw New NotSupportedException()
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/MultiplyMultiConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Ks.Common.MVVM.Converters
{
    public class MultiplyMultiConverter : IMultiValueConverter
    {
        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
        {
            var R = parameter == null ? 1 : System.Convert.ToDouble(parameter);
            for (var I = 0; I < values.Length; I++)
                R *= System.Convert.ToDouble(values[I]);
            return R;
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture)
        {
            throw new NotSupportedException();
        }
    }
}
